Replaced NApp's  publish and install functionality with GitHub (#13)
kytos-ng/documentation: tutorials/napps/publishing_your_napp.rst

Overview
********

On this tutorial you will learn how to publish your NApp to a GitHub

> repository in addition to that, share you NApp with the Kytos Community and also make

Reviewer: ```suggestion
repository. In addition to that, share you NApp with the Kytos Community and also make
```

What you will learn
===================

* How to publish your NApp on the |github|_
* How to install a NApp from |github_kytos|_

************
Introduction
************

Now that you have learned how to install the environment and how to create your
own NApp, you can publish it on |github|_.

Before proceeding to the next section of this tutorial, go to |github_signup|_ sign_up page 
in order to create a github account. After you submit with the required fields, you will be asking
to verify your email with text message sent to your email, After entering the code you will be

> redirected to github home page.

Reviewer 2: Same comment for the `username` below:

```
- **username**: String with the username of the NApp creator. While uploading your
  NApp this field must match your username on the napps server.
```

> redirected to github home page.

Reviewer 2: @krishna4041 check out this comment here, let's adapt the part where it still mentions about the username of the napps server

> redirected to github home page.

Author: Resolved.

What you will learn
===================

* How to publish your NApp on the |github|_
* How to install a NApp from |github_kytos|_

************
Introduction
************

Now that you have learned how to install the environment and how to create your
own NApp, you can publish it on |github|_.

Before proceeding to the next section of this tutorial, go to |github_signup|_ page

> in order to create a github account. After you submit with the required fields, you will be asking

Reviewer: ```suggestion
in order to create a github account. After you submit with the required fields, you will be asked
```

> to verify your email with text message sent to your email, After entering the code you will be

Reviewer: ```suggestion
to verify your email with text message you received. After entering the code you will be
```

> redirected to github home page.

Reviewer: ```suggestion
redirected to GitHub home page.
```

   - at least three characters
  --------------------------------------------------------------

The first question is related to your **username** (username should be same as your github account username).
The second question is the name of your NApp, and, for now,

> we will just use ``<name>`` as NApp name. The third question is related

Reviewer 2: Krishina, let's update this phrase to to avoid users getting confused with he ``<name>`` as a literal name:

The second question is the name of your NApp  ``<name>``, feel free to use something like ``my_first_napp`` or ``hello_world` for now.

> we will just use ``MyfirstNapp`` as NApp name. The third question is related

Reviewer 2: @krishna4041 let's make this `MyfirstNapp` snake_case as the rest of the docs typically follows. Could you find all the occurrences and replace it with `my_first_napp`?

==========

Among other things, the **README.rst** will be presented as the main content of

> the NApp page on |github|_  (https://github.com/<username>/<nappName>).

Reviewer 2: Maybe we can keep the `<name>` to reflect the same term that the kytos-util cli uses and to be aligned with the previous paragraph too

******************

Your NApp is now ready to be uploaded. 

> Before that, we need to create a repository with NApp name on github. 

Reviewer 2: we -> you

After creating repository follow the below commands to initialize and push your NApp to github.

.. code-block:: console

  $ cd ~/tutorials/<username>/my_first_napp
  $ git init

>   $ git remote add origin https://github.com/<yourGithubAccountName>/<repoName>.git

Reviewer 2: I would suggest `yourGithubAccountName` -> `username` and  `repoName` -> `name`, to use the same naming consistency throughout the docs

  $ git add .
  $ git push -u origin master

You will be asked to input your username and password of github, here password indicates generated token from github, View this |github_token|_ to generate your token.

  Enter username: <username>
  
  Enter password: <token>

After finishing the above steps your napp will be pushed to github and your will be able to view it in your repository.

> Follow Instructions in this |github_topic|_ to include topic name 'kytos-ng' to your repository.

Reviewer 2: Follow the instructions on...

Reviewer 2: Let's use the topic `kytos-napp` for the NApps instead of `kytos-ng`, we'll use `kytos-ng` only for the platform itself, that way NApps only use `kytos-napp`

Your NApp is now uploaded. You can see your NApp on the web: https://github.com/topics/kytos-napp

************************************
Search for and Install a remote NApp
************************************

> Now that you have published your NApp, you can find your NApp |kytos_page|_, you can also look for other NApps published.

Reviewer 2: ```suggestion
Now that you have published your NApp, you can find your NApp on its GitHub page. Also, you can search and discover other NApps that have been published on GitHub |kytos_page|_
```

I think this new updated phrase emphasizes more how users can still search for it, let me know what you think or if you have other suggestions

Author: May be GitHub has paginated functionality, I don't see a search box in the Github topics window, However we are indicating the end users to find there Napp at Github topic page, do you suggest any other phrase ?

Reviewer 2: Yeah, I think it'll be helpful to highlight both things here, users can find their NApps directly on its page aka http://github.com/<username>/my_first_napp, and then the topic itself even though they can be found there, this is meant for discoverability, that's the goal we're trying to achieve with the GitHub `topic` here, just so users could find other NApps by browsing the NApps, that's wy I suggested that phrase in my previous comment, feel free to reword differently, as long as we can convey both two points here that should be ok.

Author: Ok.

* Select the napp you want to install

> * Clone the NApp you want to install by Folowing below commands

Reviewer 2: ```suggestion
* Clone the NApp you want to install by following below commands
```

.. code-block:: console

>   $ git clone https://github.com/<yourName>/<repoName>.git

Reviewer 2: Same as previous comment `username` and `name`

>   $ cd repoName

Reviewer 2: ```suggestion
  $ cd <name>
```

******************

Your NApp is now ready to be uploaded. 
Before that, you need to create a repository with NApp name on github.

> After creating repository follow the below commands to initialize and push your NApp to github.

Reviewer: ```suggestion
After creating repository, follow the below commands to initialize and push your NApp to github.
```

.. code-block:: console

  $ cd ~/tutorials/<username>/my_first_napp
  $ git init
  $ git remote add origin https://github.com/<username>/my_first_napp.git
  $ git add .
  $ git commit -m "<you commit message>"
  $ git push -u origin master

You will be asked to input your username and password of github, here password indicates generated token from github, View this |github_token|_ to generate your token.

  Enter username: <username>

  Enter password: <token>

> After finishing the above steps your napp will be pushed to github and your will be able to view it in your repository.

Reviewer: ```suggestion
After finishing the above steps your NApp will be pushed to github and your will be able to view it in your repository.
```

Follow the instructions on this |github_topic|_ to include topic name 'kytos-napp' to your repository.

Your NApp is now uploaded. You can see your NApp on the web: https://github.com/topics/kytos-napp

************************************
Search for and Install a remote NApp
************************************

> Now that you have published your NApp, you can find your NApp at |kytos_page|_, you can also look for other NApps published.

Reviewer: ```suggestion
Now that you have published your NApp, you can find it at |kytos_page|_, where you can also look for other NApps published.
```

* Select the napp you want to install
* Clone the NApp you want to install by folowing below commands

.. code-block:: console

  $ git clone https://github.com/<username>/my_first_napp.git
  $ cd my_first_napp
  $ python setup.py develop

> Now that your NApp is Intalled you can view in your console by using below command

Reviewer: ```suggestion
Now that your NApp is Intalled you can view it in your console by using below command
```

   [i-]  | kytos/of_l2ls             | An L2 learning switch application for OpenFlow swi...
   [i-]  | kytos/of_lldp             | Discovers switches and hosts in the network using ...
   [i-]  | kytos/of_stats            | Provide statistics of openflow switches.

>    [ie]  | Your NApp                 | Description of your Napp

Reviewer 2: `Your NApp` here it should be `<username>/<name>`
